rasterio/_copy.pyx

```diff
@@ -1,48 +1,76 @@
 """Raster copying."""
 
+
 include "gdal.pxi"
 
 import logging
-import os
-import os.path
 
-from rasterio._err cimport exc_wrap_int, exc_wrap_pointer
+from rasterio._io cimport DatasetReaderBase
+from rasterio._err cimport exc_wrap_pointer
+from rasterio.env import ensure_env
 
 
 log = logging.getLogger(__name__)
 
 
-cdef class RasterCopier:
+@ensure_env
+def copy(src, dst, driver='GTiff', strict=False, **creation_options):
```

Comment:
As it turns out, I haven't used `copy()` as much as I expected I would. I'm willing to follow your use cases and instincts here. Removing the class is okay with me.